int8 take_place_x
int8 take_place_y
int8 put_place_x
int8 put_place_y
---
float64 take_coord_x
float64 take_coord_y
float64 take_coord_z
float64 put_coord_x
float64 put_coord_y
float64 put_coord_z